packages/ui/src/date-picker/DateRangePicker.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState, useEffect } from 'react';
+import { useStyles } from './DateRangePicker.styles';
+import {
+  DatePickerBase,
+  DatePickerBaseProps,
+  DatePickerBaseInputComponent,
+  DatePickerBaseInputComponentProps,
+} from './DatePickerBase';
+
+export type DateRangePickerValue = Date[];
+export type DateRangePickerInputComponentProps = DatePickerBaseInputComponentProps<
+  DateRangePickerValue
+>;
+
+export interface DateRangePickerProps extends DatePickerBaseProps {
+  value: DateRangePickerValue;
+  InputComponent: DatePickerBaseInputComponent<DateRangePickerInputComponentProps>;
+  onChange: (value: DateRangePickerValue) => void;
+}
+
+export function DateRangePicker({ value, onChange, ...props }: DateRangePickerProps) {
+  const { firstDayOfRange, lastDayOfRange, ...classNames } = useStyles();
+  const [pickingDateType, setPickingDateType] = useState<'start' | 'end'>('start');
+  const [startDate, setStartDate] = useState(value[0]);
+  const [endDate, setEndDate] = useState(value[1]);
```

Comment:
`value` changes do not sync back with `state`, do we really need local `state` here? maybe we can use props directly from the parent